### PR TITLE
make flatten return `Err`s instead of `Value::Error`s

### LINKED ISSUE #6805: Data access of $in replaces prior error messages with another error
### Describe the bug

Similar to #6763, if `$in` has become an error, attempting to access data from it causes a different error to occur (preventing the first error from being presented to the user).

### How to reproduce

```
ls | group-by type | flatten | $in.dir
Error: nu::shell::incompatible_path_access (link)

  × Data cannot be accessed with a cell path
   ╭─[entry #35:1:1]
 1 │ ls | group-by type | flatten | $in.dir
   ·                                    ─┬─
   ·                                     ╰── error doesn't support cell paths
   ╰────
```

### Expected behavior

I expect the original error to still be presented regardless of the occurrence of `$in.dir`.

### Screenshots

_No response_

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.69.1                                          |
| branch             |                                                 |
| commit_hash        | 8a5abc7afc28a36908a32f1852127e2a5170b461        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.64.0 (a55dd71d5 2022-09-19)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.64.0 (387270bc7 2022-09-16)             |
| pkg_version        | 0.69.1                                          |
| build_time         | 2022-09-28 02:58:08 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by merelymyself**
# Description

Fixes #6805. Ideally we would want to do the same for all other commands.

# Tests

Make sure you've done the following:

- [ ] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

# Documentation

- [ ] If your PR touches a user-facing nushell feature then make sure that there is an entry in the documentation (https://github.com/nushell/nushell.github.io) for the feature, and update it if necessary.

### CODE REVIEW

File: crates/nu-command/src/filters/flatten.rs [hunk 1]
```diff
@@ -116,12 +117,17 @@ fn flatten(
...
-    input
-        .flat_map(
-            move |item| flat_value(&columns, &item, tag, flatten_all),
-            engine_state.ctrlc.clone(),
-        )
-        .map(|x| x.set_metadata(metadata))
+    let input_iter = input.into_interruptible_iter(engine_state.ctrlc.clone());
+    let mut iter_val_vec = Vec::new();
+    for item in input_iter {
+        match flat_value(&columns, &item, tag, flatten_all) {
```
Comment (jntrnr) [inline crates/nu-command/src/filters/flatten.rs:123]: This looks like it switches the lazy stream into not being lazy. As much as possible, the pipelines should work like normal iterators so values are computed only when needed.

Is there a different way we can solve it that doesn't compute the whole stream first?

Comment (merelymyself) [reply crates/nu-command/src/filters/flatten.rs:123]: I think I'm seeing the problem here. When the input is being processed, it gets streamed out with `PipelineData` already being returned. However, if a ShellError is encountered, there is no way for `run` to return an `Err` value. 

I'm not sure if there's an easy fix to this, but I'll keep trying.

**Comment by merelymyself** [comment]
Going to need to think a bit more about this problem. Closing this PR for now.
